aiida_common_workflows/workflows/bands/castep/workchain.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+"""Implementation of `aiida_common_workflows.common.relax.workchain.CommonRelaxWorkChain` for CASTEP."""
+from logging import getLogger
+
+from aiida.engine import calcfunction
+from aiida.orm import Float
+from aiida.plugins import WorkflowFactory
+
+from ..workchain import CommonBandsWorkChain
+from .generator import CastepCommonBandsInputGenerator
+
+__all__ = ('CastepCommonBandsWorkChain',)
+
+
+@calcfunction
+def get_fermi_energy(bands):
+    """Extract the Fermi energy from the BandsData output"""
+    efermi = bands.get_attribute('efermi')
+    if isinstance(efermi, list):
+        efermi = efermi[0]
```

Comment:
This is a good point. Should we discuss whether the `fermi_energy` output should be a `Float` or a `List[float]` depending on polarization? Think this would be a common "problem" across implementations

Reply:
I will put in the list for discussion tomorrow

Reply:
An additional issue is that some code may not have a well defined single efermi energy if there are two spin channels (is such quantity itself well-defined at all?). Not sure if this is a limitation across the board or not. I know CASTEP always give two if there are two spin channels. VASP gives ones fermi energy with spin polarised calculation I think, but it may give a wrong value if spin is constrained.